front/components/assistant_builder/ActionScreen.tsx

```diff
@@ -334,142 +339,138 @@ export default function ActionScreen({
 
         {getActionType(builderState.actionMode) === "USE_DATA_SOURCES" && (
           <>
-            <div className="flex flex-row items-center space-x-2">
-              <div className="text-sm font-semibold text-element-900">
-                Method:
-              </div>
-              <DropdownMenu>
-                <DropdownMenu.Button>
-                  <Button
-                    type="select"
-                    labelVisible={true}
-                    label={
-                      SEARCH_MODE_SPECIFICATIONS[
-                        getSearchMode(builderState.actionMode)
-                      ].label
-                    }
-                    icon={
-                      SEARCH_MODE_SPECIFICATIONS[
-                        getSearchMode(builderState.actionMode)
-                      ].icon
+            {noDataSources ? (
+              <ContentMessage
+                title="You don't have any Data source available"
+                variant="warning"
+              >
+                <div className="flex flex-col gap-y-3">
+                  {(() => {
+                    switch (owner.role) {
+                      case "admin":
+                        return (
+                          <div>
+                            <strong>
+                              Visit the "Connections", "Websites" and "Folders"
+                              sections in the Build panel to add new data
+                              sources.
+                            </strong>
+                          </div>
+                        );
+                      case "builder":
+                        return (
+                          <div>
+                            <strong>
+                              Only Admins can activate Connections.
+                              <br />
+                              You can add Data Sources by visiting the
+                              "Websites" and "Folders" sections in the Build
+                              panel.
+                            </strong>
+                          </div>
+                        );
+                      case "user":
+                        return (
+                          <div>
+                            <strong>
+                              Only Admins and Builders can activate Connections,
```

Comment:
nit: copy change. feels like saying to the user "You don't have data sources, and you can't do anything about it, cheers"

I'd phrase it "Contact a Dust Admin to add connections, URLs or folders". (Same for builder)

Reply:
just seeing this copy is from former code; but good occasion for a quick improvement IMHO

Reply:
tweaked